kivakit-configuration/src/main/java/com/telenav/kivakit/configuration/BaseComponent.java

```diff
@@ -0,0 +1,192 @@
+     * By default, this is the global settings store.
+     */
+    public void addDeployment(final Deployment deployment)
+    {
+        settings().addDeployment(deployment);
+    }
+
+    /**
+     * Adds the given settings object to the settings registry for this application. By default, this is the global
+     * settings store.
+     */
+    public void addSettings(final Object settings)
+    {
+        settings().add(settings);
+    }
+
+    /**
+     * Adds the identified settings object to the settings registry for this application. By default, this is the global
+     * settings store.
+     */
+    public void addSettings(final Object configuration, final InstanceIdentifier instance)
+    {
+        settings().add(configuration, instance);
+    }
+
+    /**
+     * @return Any registered object of the given type with the given instance identifier in the global lookup registry
+     */
+    public <T> T lookup(final Class<T> type, final String instance)
+    {
+        return registry().find(type, instance);
+    }
+
+    /**
+     * @return Any registered object of the given type with the given instance identifier in the global lookup registry
+     */
+    public <T> T lookup(final Class<T> type, final InstanceIdentifier instance)
+    {
+        return registry().find(type, instance);
+    }
+
+    /**
+     * @return Any registered object of the given type with the given instance identifier in the global lookup registry
+     */
+    public <T> T lookup(final Class<T> type, final Enum<?> instance)
+    {
+        return registry().find(type, instance);
+    }
+
+    /**
+     * @return The object of the given type from the application's lookup {@link Registry}, if any.
+     */
+    public <T> T lookup(final Class<T> type)
+    {
+        return registry().find(type);
+    }
+
+    /**
+     * Registers the given singleton object in the global lookup registry
+     */
+    public <T> T register(final T object)
+    {
+        return registry().add(object);
+    }
+
+    /**
+     * Registers the specified instance of the given object's type in the global lookup registry
+     */
+    public <T> T register(final T object, final String instance)
+    {
+        return registry().add(object, instance);
+    }
+
+    /**
+     * Registers the specified instance of the given object's type in the global lookup registry
+     */
+    public <T> T register(final T object, final InstanceIdentifier instance)
+    {
+        return registry().add(object, instance);
+    }
+
+    /**
+     * Registers the specified instance of the given object's type in the lookup
+     */
+    public <T> T register(final T object, final Enum<?> instance)
+    {
+        return registry().add(object, instance);
+    }
+
+    /**
+     * @return The lookup registry for this application
+     */
+    public Registry registry()
+    {
+        return Registry.of(this);
```

Comment:
According to previous code, seems it should be Registry.global().  Although current Registry.of() also call Registry.global(), I am not sure it might be changed as object specified Registry instead of global.

Reply:
The design here is so that we can provide a hook in Registry so that it would be possible to provide different registries based on the requesting object. One can imaging a complex requirement where some objects are in one container and others are in a different container. When an object in one container looks up an object, the BaseComponent.registry() method is one place where an override could occur. Another would be in Registry.of(). For now, Registry.of() just returns the global registry, but in the future, we can add a mechanism for returning other registries. So, the preferred method to use is Registry.of(), which is what BaseComponent call. Make sense?

Reply:
Got it.